### DISCUSSION TITLE
Timeout while loading URL when trying to set up a feed

### DISCUSSION THREAD

**Discussion by snigehere** [tsightler/ring-mqtt#435] [Support Q&A]
Hi
I have a ring alarm and 5 cameras - 2 stickup, one floodlight, one spotlight and one doorbell.

I have successfully added the generic camera integrations, and configured them, for the 3 main cameras.  These show successfully in my dashboards - I am now trying to get the two stickup cameras to show the live stream when they detect motion. I have obtained the stream url's from the info record and created the generic camera.  I can successfully add the snapshot url and am able to see the snapshot updating but when I try to add the stream source to the configuration eg:  (note: data removed and replaced with "*" so I am not sharing personal info) 
rtsp://*******:******@*********-ring-mqtt:8554/54********aa_live 
I get an error "Timeout while loading URL" and in the home assistant log I see
2022-08-23 22:52:11.785 ERROR (stream_worker) [homeassistant.components.stream.stream.test_stream] Error from stream worker: Error opening stream (HTTP_NOT_FOUND, Server returned 404 Not Found) rtsp://****:****@********-ring-mqtt:8554/*******_live

I don't see any transaction in the "Ring-MQTT with Video Streaming" add on log when the add fails but when the camera detects motion I can see transactions in the log showing that the stream is unmasked

It seems odd that the processes worked for 3 cameras but not the last 2 - is there some restrictions related to battery stickup cams?  or are there different configuration items I need to select?

Any help would be appreciated

**Comment by tsightler** [comment]
Stickup cams should work fine, I have two of them myself, although they are wired, but there's no expected difference in behavior between them.  The fact that the error is 404 implies that the URL is incorrect.  Note that the created paths are logged during startup in the ring-mqtt log so you can verify that the path portion of the URL is 100% correct.

**Reply by snigehere** [reply]
Thanks for the fast response ... I restarted the addon and can see the device discovery in the log:
2022-08-23T23:12:45.640Z ring-mqtt   New device: shed (9c76131666ab)
2022-08-23T23:12:45.640Z ring-mqtt               cocoa_camera

The number for the device corresponds to the one in the rtsp statement (with _live added to the end) which aligns with the one shown in the _info record.  I compared it to the rtsp statement from one of the working cameras and its exactly the same other than the device ID

rtsp://****:****@03*****-ring-mqtt:8554/9c76131666ab_live ... the mqtt id and password match the working one as well. 

Does the log get written out somewhere that I can access?  Its quite short and the start-up fills it up quickly so I didn't get it all.

**Reply by tsightler** [reply]
The full logs can be accessed via the standard "docker logs" command you'll need to have an SSH addon running with protection mode disabled to be able to run docker commands on Home Assistant OS).  High level instructions are here:

https://github.com/tsightler/ring-mqtt-ha-addon/blob/main/GET-LOGS.md

**Reply by tsightler** [reply]
To be clear, the example there is how to dump logs to a file, if you just want to view them you can pipe them to whatever tool you like.

**Comment by jlchavis0844** [comment]
Bumping this topic. I am having a similar issue where my generic camera cannot open my rtsp feeds. I enabled external port access and tried loading the feed via VLC and it also times out.

Error from stream worker: Error opening stream (HTTP_NOT_FOUND, Server returned 404 Not Found) rtsp://****:****@03cabcc9-ring-mqtt:8554/f045dafa9e28_live

**Reply by tsightler** [reply]
HTTP_NOT_FOUND is not a timeout, you are using an invalid URL.  Assuming you have not enabled any username/password for the livestream then you should just drop the username/password portion of the URL completely:

rtsp://03cabcc9-ring-mqtt:8554/f045dafa9e28_live
